### PR TITLE
hysteria: 1.2.0 -> 1.2.1

### PR DESCRIPTION
**Pull request by oluceps**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/hysteria/default.nix [hunk 1]
```diff
@@ -1,21 +1,32 @@
 { lib
 , fetchFromGitHub
-, buildGoModule
+, buildGo118Module
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hysteria/default.nix:3]:
```suggestion
, buildGoModule
```

File: pkgs/tools/networking/hysteria/default.nix [hunk 2]
```diff
@@ -1,21 +1,32 @@
...
 }:
-buildGoModule rec {
+buildGo118Module rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/hysteria/default.nix:5]:
```suggestion

buildGoModule rec {
```
In the meantime go 1.18 is the default.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/hysteria/default.nix (hunk 2 above)
Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix:5]: It used to be like this, and here was the error that appeared before I changed `buildGoModule` into `buildGo118Module`. Note the penultimate line, I don't know if the version incompatible is the reason, but everything is solved after specifying the 118 version.
```
....
github.com/spf13/viper
github.com/mholt/acmez
github.com/lucas-clemente/quic-go/http3
github.com/HyNetwork/hysteria/pkg/transport
github.com/caddyserver/certmagic
github.com/HyNetwork/hysteria/pkg/core
github.com/prometheus/client_golang/prometheus/promhttp
# github.com/caddyserver/certmagic
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/certmagic.go:408:11: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/certmagic.go:428:13: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/config.go:68:66: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/config.go:1115:80: undefined: any
../go/pkg/mod/github.com/caddyserver/certmagic@v0.17.1/solvers.go:699:7: undefined: any
note: module requires Go 1.18
github.com/folbricht/routedns
....
```

Here is the [full log](https://gist.github.com/oluceps/763726f8f8deb17f4e79eabe83ddfb11)

Comment (SuperSandro2000) [reply pkgs/tools/networking/hysteria/default.nix:5]: buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix:5]: > buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

But the only modification that caused build failed is `buildGoModule`, that's weird

Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix:5]: > > buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367
> 
> But the only modification that caused build failed is `buildGoModule`, that's weird

And seems that some dependence reports require Go 1.18

Comment (oluceps) [reply pkgs/tools/networking/hysteria/default.nix]: > buildGoModule is an alias to buildGo118Module https://github.com/NixOS/nixpkgs/blob/8a810b1a1611e7deaafc7b1bded2976b2a6ee5c2/pkgs/top-level/all-packages.nix#L22367

Alright, it passed `nixpkgs-review rev HEAD`, works well with `buildGoModule`. And still failed to build with `nix-build -E 'with import <nixpkgs> {}; callPackage ./default.nix {}'`

**Pull request merged**